jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsSubsystemParser_2_0.java

```diff
@@ -0,0 +1,132 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.jaxrs;
+
+import static org.jboss.as.controller.parsing.ParseUtils.requireNoAttributes;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Pattern;
+
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.controller.parsing.ParseUtils;
+import org.jboss.as.controller.persistence.SubsystemMarshallingContext;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.jboss.staxmapper.XMLElementReader;
+import org.jboss.staxmapper.XMLElementWriter;
+import org.jboss.staxmapper.XMLExtendedStreamReader;
+import org.jboss.staxmapper.XMLExtendedStreamWriter;
+
+public class JaxrsSubsystemParser_2_0 implements XMLStreamConstants, XMLElementReader<List<ModelNode>>, XMLElementWriter<SubsystemMarshallingContext> {
+
+   private static final Pattern PATTERN_LIST = Pattern.compile(".+(,.+)*");
+   private static final Pattern PATTERN_PROPERTY_LIST = Pattern.compile(".+:.+(,.+:.+)*");
+
+   /**
+    * {@inheritDoc}
+    */
+   @Override
+   public void readElement(final XMLExtendedStreamReader reader, final List<ModelNode> list) throws XMLStreamException {
+       // Require no attributes or content
+       requireNoAttributes(reader);
+       list.add(Util.createAddOperation(PathAddress.pathAddress(JaxrsExtension.SUBSYSTEM_PATH)));
+
+       //Read the children
+       while (reader.hasNext() && reader.nextTag() != END_ELEMENT) {
+           if (!reader.getLocalName().equals("context-parameters")) {
+               throw ParseUtils.unexpectedElement(reader);
+           }
+           while (reader.hasNext() && reader.nextTag() != END_ELEMENT) {
+               if (reader.isStartElement()) {
+                   readContextParameter(reader, list);
+               }
+           }
+       }
+   }
+
+   private void readContextParameter(XMLExtendedStreamReader reader, List<ModelNode> list) throws XMLStreamException {
+      if (!reader.getLocalName().equals("context-parameter")) {
+          throw ParseUtils.unexpectedElement(reader);
+      }
+
+      String name = reader.getAttributeValue(0);
+      String value = reader.getAttributeValue(1);
+      if (PATTERN_PROPERTY_LIST.matcher(value).matches() || PATTERN_LIST.matcher(value).matches()) {
+         String[] ss = value.split(",");
+         ModelNode modelNode = new ModelNode();
+         for (String s : ss) {
+            modelNode.add(s.trim());
+         }
+         JaxrsContextParamHandler.getContextParameters().put(name, modelNode);
```

Comment:
This should not deal with JaxrsContextParameterHandler directly.

Instead the add handler (see my other point about putting these into a child resource of the subsystem) should be enhanced to deal with all these new parameters, so they get added to the model (they don't at the moment) and populate the context parameters cache in the same way as your write attribute handler.

I was getting quite confused why your subsystem test was working while not being able to see values persisted in the xml show up in the model of a restarted server :)